src/generator/operations.ts

```diff
@@ -427,10 +427,15 @@ function createProtocolRequest(client: string, op: Operation, imports: ImportMan
   const mediaType = getMediaType(op.requests![0].protocol);
   if (mediaType === 'JSON' || mediaType === 'XML') {
     const bodyParam = values(aggregateParameters(op)).where((each: Parameter) => { return each.protocol.http!.in === 'body'; }).first();
+    // adding this variable to control whether a 'options.' needs to be added before optional body parameters
+    let setOptionsPrefix = false;
     // default to the body param name
     let body = bodyParam!.language.go!.name;
     if (bodyParam!.schema.type === SchemaType.Constant) {
       // if the value is constant, embed it directly
+      if (!bodyParam!.required) {
```

Comment:
Nit: Can be replaced with `setOptionsPrefix = !bodyParam!.required` to eliminate nested `if` block.